internal/interpreter/api.rs

```diff
@@ -579,8 +579,18 @@ impl ComponentCompiler {
                 Box<dyn core::future::Future<Output = Option<std::io::Result<String>>>>,
             > + 'static,
     ) {
-        self.config.open_import_fallback =
-            Some(Rc::new(move |path| file_loader_fallback(Path::new(path.as_str()))));
+        let file_loader_fallback = Rc::new(file_loader_fallback);
+
+        let open_import_fallback = move |file_name: String| -> core::pin::Pin<
+            Box<dyn core::future::Future<Output = Option<std::io::Result<(String, Option<i32>)>>>>,
+        > {
+            let flfb = file_loader_fallback.clone();
+            Box::pin({
+                async move { flfb(&PathBuf::from(file_name)).await.map(|r| r.map(|c| (c, None))) }
+            })
+        };
```

Comment:
I think we can manage without putting the file_loader_fallback in a extra RC thereby saving one allocation and one indirection:

```suggestion
        let open_import_fallback = move |file_name: String| -> core::pin::Pin<
            Box<dyn core::future::Future<Output = Option<std::io::Result<(String, Option<i32>)>>>>,
        > {
            let flfb = file_loader_fallback(&Path::new(file_name.as_str()));
            Box::pin(async move { flfb.await.map(|r| r.map(|c| (c, None))) })
        };
```

(untested)

Reply:
You are right! Wonderful.